src/main/java/io/odpf/kafkaclient/ClientFactory.java

```diff
@@ -1,4 +1,31 @@
 package io.odpf.kafkaclient;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
 public abstract class ClientFactory {
+    private Properties properties;
+
+    public final void configure(File file) throws IOException {
```

Comment:
why final ?

Reply:
so that subclasses do not change the behaviour of extracting properties from the properties file